reconcile/aus/ocm_addons_upgrade_scheduler_org.py

```diff
@@ -117,41 +116,58 @@ def get_state_for_org_spec_per_addon(
         org_upgrade_spec, fetch_current_state
     )
     result = []
-    for addon_id in set(a["addon_id"] for a in desired_state):
-        addon_current_state = [c for c in current_state if c["addon_id"] == addon_id]
-        addon_desired_state = [d for d in desired_state if d["addon_id"] == addon_id]
+    for addon_id in set(
+        a.addon_id
+        for a in desired_state
+        if isinstance(a, aus.ConfiguredAddonUpgradePolicy)
+    ):
+        addon_current_state = [
+            c
+            for c in current_state
+            if isinstance(c, aus.AddonUpgradePolicy) and c.addon_id == addon_id
+        ]
+        addon_desired_state = [
+            d
+            for d in desired_state
+            if isinstance(d, aus.ConfiguredAddonUpgradePolicy)
+            and d.addon_id == addon_id
+        ]
         result.append(
             AUSOrgAddonUpgradeState(
                 addon_id=addon_id,
                 current_state=addon_current_state,
-                desired_state=addon_desired_state,
+                upgrade_policies=addon_desired_state,
             )
         )
     return ocm_map, result
 
 
 def calculate_diff(
-    addon_current_state: list[dict[str, Any]],
-    addon_desired_state: list[dict[str, Any]],
+    addon_current_state: list[aus.AbstractUpgradePolicy],
+    addon_upgrade_policies: list[aus.ConfiguredUpgradePolicy],
     ocm_map: OCMMap,
     version_data_map: dict[str, VersionData],
     addon_id: str = "",
-) -> list[dict[str, str]]:
+) -> list[aus.AbstractUpgradePolicy]:
     diffs = aus.calculate_diff(
         addon_current_state,
-        addon_desired_state,
+        addon_upgrade_policies,
         ocm_map,
         version_data_map,
         addon_id=addon_id,
     )
-    for c in addon_current_state:
-        if addon_id == c["addon_id"] and c["schedule_type"] == "automatic":
+    for c in [p for p in addon_current_state if isinstance(p, aus.AddonUpgradePolicy)]:
+        if addon_id == c.addon_id and c.schedule_type == "automatic":
```

Comment:
Since there is a for loop anyway, can skip the list comprehension and move conditions together like:

```python
for c in addon_current_state:
    if isinstance(c, aus.AddonUpgradePolicy) and addon_id == c.addon_id and c.schedule_type == "automatic":
```

Reply:
:heavy_check_mark: